### PR TITLE
Add Versioning for v5.0 plus versioning checks for 4.3

### LINKED ISSUE #174: Add versioning 50
For the 5.0 milestone

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Fixes #174 

Built following the wiki so hopefully I didn't miss any steps! 😄

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@adecler I was getting this on the local set up too, but I couldn't fathom where it's coming from? Have you seen it before?

image

**Comment by adecler** [comment]
@FraserGreenroyd , yes, this means there is a line in the json file you produce that contains an object of type test (It should be around line 67 if I remember correctly). The easiest way to find it is to run the test from Grasshopper and check which index is producing the fail. You can safely remove that line.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@adecler finally got the bot to like this PR - should be good for a review :smile:

File: .ci/code/Versioning_Test/Verify/FromJson.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static partial class Verify
         public static TestResult FromJsonDatasets()
         {
             string testFolder = @"C:\ProgramData\BHoM\Datasets\TestSets\Versioning";
-            List<string> versions = new List<string> { "4.2", "4.1", "4.0", "3.3" };
+            List<string> versions = new List<string> { "4.3", "4.2", "4.1", "4.0", "3.3" };
```
Comment (adecler) [inline .ci/code/Versioning_Test/Verify/FromJson.cs:50]: We might want to keep that list to a minimum based on our analytics of version usage. 
Happy with it staying this way for now though.

**Review by adecler — APPROVED** [review]
Thanks @FraserGreenroyd ,

Looks good to me. 
Just made a note on the list of versions to test but just something to keep in mind with no immediate action.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**